src/api/wallet/WalletApi.ts

```diff
@@ -515,6 +521,19 @@ export class WalletApiImpl implements WalletApi {
 
   /* ****************      Private Functions      **************** */
 
+  private _setProviderInfo(): void {
+    // this can get expensive depending on the number and complexity of checks in getProviderInfo
+    // so retrigger only on connect/disconnect
+    this._providerInfo = getProviderInfo(this._provider)
+
+    if (this._providerInfo && isWalletConnectProvider(this._provider) && this._provider.wc.peerMeta) {
```

Comment:
what is the `peerMeta` for?

Reply:
Probably lots of stuff. It's where WC stores [wallet info](https://github.com/WalletConnect/walletconnect-monorepo/blob/efd31442ca7251ccdef3ca4d832a8f01276bf74d/packages/helpers/types/index.d.ts#L183)
There's also `clientMeta` which seems to be the same